packages/app/src/specs/SpecsListBanners.cy.tsx

```diff
@@ -73,58 +68,97 @@ describe('<SpecsListBanners />', () => {
+
+      loginConnectStore.setCypressFirstOpened(Date.now() - interval('3 days'))
+
+      cy.mountFragment(SpecsListBannersFragmentDoc, {
+        render: (gqlVal) => <SpecsListBanners gql={gqlVal} />,
+      })
 
       cy.get(`[data-cy="${bannerTestId}"]`).should('not.exist')
     })
 
     it('should not render when previously-dismissed', () => {
-      mountWithState(gql, {
-        ...stateWithFirstOpenedDaysAgo(4),
-        banners: {
-          [bannerId]: {
-            dismissed: Date.now(),
-          },
+      const loginConnectStore = useLoginConnectStore()
+
+      loginConnectStore.setBannersState({
+        [bannerId]: {
+          dismissed: Date.now(),
         },
       })
 
+      cy.mountFragment(SpecsListBannersFragmentDoc, {
+        render: (gqlVal) => <SpecsListBanners gql={gqlVal} />,
+      })
+
       cy.get(`[data-cy="${bannerTestId}"]`).should('not.exist')
     })
 
     context('banner conditions are met and when cypress use >= 4 days', () => {
       beforeEach(() => {
+        const loginConnectStore = useLoginConnectStore()
+
+        loginConnectStore.setCypressFirstOpened(Date.now() - interval('4 days'))
+
         cy.stubMutationResolver(UseCohorts_DetermineCohortDocument, (defineResult) => {
           return defineResult({ determineCohort: { __typename: 'Cohort', name: 'foo', cohort: 'A' } })
         })
       })
 
       it('should render when not previously-dismissed', () => {
-        mountWithState(gql, stateWithFirstOpenedDaysAgo(4))
+        cy.mountFragment(SpecsListBannersFragmentDoc, {
+          render: (gqlVal) => <SpecsListBanners gql={gqlVal} />,
+        })
+
+        const bannerTrueUserConditions = {
+          'login-banner': [],
+          'create-organization-banner': ['isLoggedIn', 'isOrganizationLoaded'],
+          'connect-project-banner': ['isLoggedIn', 'isMemberOfOrganization'],
+          'record-banner': ['isLoggedIn', 'isMemberOfOrganization'],
+        } as const
+
+        const bannerTrueProjectConditions = {
+          'login-banner': [],
+          'create-organization-banner': [],
+          'connect-project-banner': ['isConfigLoaded'],
+          'record-banner': ['isProjectConnected', 'hasNoRecordedRuns', 'hasNonExampleSpec', 'isConfigLoaded'],
+        } as const
+        const loginConnectStore = useLoginConnectStore()
+
+        bannerTrueUserConditions[bannerTestId].forEach((status: keyof LoginConnectState['user']) => {
+          loginConnectStore.setUserFlag(status, true)
+        })
+
+        bannerTrueProjectConditions[bannerTestId].forEach((status: keyof LoginConnectState['project']) => {
+          loginConnectStore.setProjectFlag(status, true)
+        })
+
         cy.get(`[data-cy="${bannerTestId}"]`).should('be.visible')
       })
 
       it('should be preempted by spec not found banner', () => {
-        mountWithState(gql, stateWithFirstOpenedDaysAgo(4), { isSpecNotFound: true })
+        mountWithState(gql, undefined, { isSpecNotFound: true })
```

Comment:
Sorta don't love passing this `undefined`, but also don't want to touch more code in this PR to refactor this function.